common/src/edgeai_camera.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright (C) 2023 Texas Instruments Incorporated - http://www.ti.com/
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* Module headers */
+#include <common/include/edgeai_camera.h>
+#include <utils/include/ti_logger.h>
+
+namespace ti::edgeai::common
+{
+
+using namespace ti::utils;
+
+camera::camera()
+{
+    LOG_DEBUG("camera CONSTRUCTOR\n");
+}
+
+camera::~camera()
+{
+    LOG_DEBUG("multiScaler DESTRUCTOR\n");
+}
+
+int32_t camera::getConfig(int32_t chMask)
+{
+    vx_status status = VX_SUCCESS;
+
+    tiovx_sensor_module_params_init(&sensorObj);
+    sensorObj.is_interactive = 0;
+    sensorObj.ch_mask = chMask;
+
+    tiovx_sensor_module_query(&sensorObj);
+    sensorObj.sensor_index = 0; /* App works only for IMX390 2MP cameras */
```

Comment:
@s-jain1 Is there any need here to allow for configuring this to use different cameras?

Reply:
If in future support for more cameras is added than we will have to just modify here and also, it will be easy for anyone to know where to update the sensor index, in case they want to use some other sensor than IMX390

Reply:
Now this is made configurable and it comes from YAML config file. Added commit for this.